tex/generic/pgf/frontendlayer/tikz/tikz.code.tex

```diff
@@ -1899,16 +1899,36 @@
 % #1: shortcut command inside of tikzpicture
 % #2: real command name
 \def\tikzaddtikzonlycommandshortcutlet#1#2{%
-    \expandafter\def\expandafter\tikz@installcommands\expandafter{\tikz@installcommands
-        \let#1=#2%
+    \edef\tikz@installcommands{\pgfutil@unexpanded\expandafter{\tikz@installcommands}%
+        \ifdefined#1%
+            \let\expandafter\noexpand\csname tikz@orig\detokenize{#1}\endcsname=\noexpand#1%
+        \fi
+        \pgfutil@unexpanded{\let#1=#2}%
+    }%
+    \edef\tikz@uninstallcommands{\pgfutil@unexpanded\expandafter{\tikz@uninstallcommands}%
+        \ifdefined#1%
+            \let\noexpand#1\expandafter\noexpand\csname tikz@orig\detokenize{#1}\endcsname
+        \else
+            \pgfutil@unexpanded{\let#1=\pgfutil@undefined}%
+        \fi
```

Comment:
This won't work because now `\ifdefined` is executed at the point where `\tikzaddtikzonlycommandshortcutlet` is used. However at this point `#2` might not be defined *yet*, so code like this will restore the wrong thing
```latex
\tikzaddtikzonlycommandshortcutlet{\oops}{\specialsauce} % <-- \specialsauce not yet defined
\def\specialsauce{tada}
```

Reply:
`\ifdefined` always acts on `#1` so when `#2` is defined is irrelevant here and I suppose you want to say

> However at this point `#1` **(not `#2`)** might not be defined _yet_, so code like this will restore the wrong thing
> 
> ```
> \tikzaddtikzonlycommandshortcutlet{\oops}{\myspecialsauce} % <-- \oops not yet defined
> \def\oops{tada}
> ```
Or maybe you noticed in the patch to uninstallcommands, we should check if `\tikz@orig#1` is defined.

Anyway I find it suffices to just construct cs from `\csname tikz@orig ...\endcsname` before adding it to installcommands/uninstallcommands.